salt/roster/terraform.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+'''
+Dynamic roster from terraform current state
+===========================================
+
+This roster module allows you dynamically generate the roster from the terraform resources defined with `terraform-provider-salt <https://github.com/dmacvicar/terraform-provider-salt>`_.
+
+It exposes all salt_host resources with the same attributes to the salt-ssh roster, making it completely independent of the type of terraform resource, and providing the integration using terraform constructs with interpolation.
+
+Basic Example
+-------------
+
+Given a simple salt-ssh tree with a Saltfile
+
+.. code-block:: yaml
+
+    salt-ssh:
+      config_dir: etc/salt
+      max_procs: 30
+      wipe_ssh: True
+
+and etc/salt/master
+
+.. code-block:: yaml
+
+    root_dir: .
+      file_roots:
+        base:
+          - srv/salt
+      pillar_roots:
+        base:
+          - srv/pillar
+      roster: terraform
+
+In the same folder as your `Saltfile`, create terraform file with resources like cloud instances, virtual machines, etc. For every single of those that you want to manage with Salt, create a `salt_host` resource:
+
+.. code-block:: hcl
+    resource "salt_host" "dbminion" {
+      salt_id = "dbserver"
+      host = "${libvirt_domain.vm-db.network_interface.0.addresses.0}"
+      user = "root"
+      passwd = "linux"
+}
+
+You can use the count attribute to create multiple roster entries with a single definition. Please refer to the `terraform salt provider <https://github.com/dmacvicar/terraform-provider-salt>`_ for more detailed examples.
+'''
+# Import Python libs
+from __future__ import absolute_import, unicode_literals
+import logging
+import os.path
+
+# Import Salt libs
+import salt.utils.files
+import salt.utils.json
+from salt.roster.flat import RosterMatcher
```

Comment:
Can we move this to `__utils__` and then call it from a `__utils__` function, like `roster.targets()` which calls RosterMatcher, instead of having to pass in `__opts__`?